util/collections/src/garbage/atomic/counted.rs

```diff
@@ -58,43 +55,36 @@ impl GcCounter {
 	/// Increments the value in a specific slot.
 	pub fn increment(&self, current_time: u64, amount: u64) {
 		let slot_timestamp = current_time / self.gc_slot_duration.get();
-		let slot = slot_timestamp % self.num_slots;
+		let pass_number = slot_timestamp / (self.num_slots / 2); // Determine if we're in the first or second pass
```

Comment:
If `self.num_slots` is odd, we could get a small "third pass" here, where `pass_number % 2 == 0`, but it will quickly cycle back to the first pass that will go to the same slot offset.